packages/orchestration/src/exos/local-chain-account-kit.js

```diff
@@ -158,6 +177,37 @@ export const prepareLocalChainAccountKit = (
           trace('got result', result);
           return result;
         },
+        /**
+         *
+         * @param {string} validatorAddress
+         * @param {Amount<'nat'>} ertpAmount
+         * @returns {Promise<void>}
+         */
+        async undelegate(validatorAddress, ertpAmount) {
+          // TODO #9211 lookup denom from brand
+          const amount = {
+            amount: String(ertpAmount.value),
+            denom: 'ubld',
+          };
+          const { account: lca } = this.state;
+          trace('lca', lca);
+          const delegatorAddress = await E(lca).getAddress();
+          trace('delegatorAddress', delegatorAddress);
+          const [response] = await E(lca).executeTx([
+            typedJson('/cosmos.staking.v1beta1.MsgUndelegate', {
+              amount,
+              validatorAddress,
+              delegatorAddress,
+            }),
+          ]);
+          trace('undelegate response', response);
+          const { completionTime } = response;
+
+          await E(timerService).wakeAt(
+            // TODO clean up date handling once we have real data
+            dateInSeconds(new Date(completionTime)) + maxClockSkew,
```

Comment:
`completionTime` _is_ a `Date`, IIUC.

When do we address this TODO? It seems critical to correctness for this PR.

Reply:
agreed. I think we'll address before closing the stories. I'm going to work now on improving our data mocking